### DISCUSSION TITLE
Metric units

### DISCUSSION THREAD

**Discussion by sajmons** [ricktu288/ray-optics#140] [Q&A] [answered]
Is it possible to set ruller units to milimeters or centimeters? If not, can it be done?

**Comment by ricktu288** [comment, marked as answer]
In this web app, all the length scales (coordinates, ruler, focal length, etc.) are in the same arbitrary unit, which corresponds to the CSS pixels when the zoom percentage is 100. Since the wavelength is assumed to be infinitesimal in ray optics, the absolute length scale of the optical system is irrelevant in the simulation. You can freely interpret that unit to be millimeters, kilometers, or whatever. The simulation result will always be the same.

**Comment by ricktu288** [comment]
Another recent proposal is to allow a custom unit (which in practice is a physical unit like mm) which is different from the CSS pixel. Then the "mm-to-px-ratio" will become a global settings. This is essentially to decouple the unit used in the simulator and the line width of the rendering of rays and optical elements, etc. This may also be generalized to the physical unit of light intensity (and after #37 is implemented, the speed of light).

Note however that we need to distinguish the wavelength-related length unit (which has physical meaning), and the arbitrary length unit (which has no physical meaning).

Note also that some proposals of #141 and #142 may complicate things a bit, since it may not be good if the raw unit of the JSON-like editor in the UI be different from the length unit shown to the user elsewhere.

**Reply by ricktu288** [reply]
Implemented as Settings -> Length Scale.
